apps/cowswap-frontend/src/modules/swap/helpers/getAmountsForSignature.test.ts

```diff
@@ -0,0 +1,143 @@
+import { USDC_MAINNET, WETH_MAINNET } from '@cowprotocol/common-const'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { Currency, CurrencyAmount, Percent, Price, TradeType } from '@uniswap/sdk-core'
+
+import TradeGp from 'legacy/state/swap/TradeGp'
+
+import { getAmountsForSignature } from './getAmountsForSignature'
+
+interface TradeParams {
+  tradeType: TradeType
+  inputAmount: CurrencyAmount<Currency>
+  outputAmount: CurrencyAmount<Currency>
+  feeAmount: CurrencyAmount<Currency>
+}
+
+function getTrade(params: TradeParams): TradeGp {
+  const { tradeType, inputAmount, outputAmount, feeAmount } = params
+
+  const isExactInput = tradeType === TradeType.EXACT_INPUT
+
+  const executionPrice = new Price({
+    baseAmount: inputAmount,
+    quoteAmount: outputAmount,
+  })
+
+  return new TradeGp({
+    inputAmount,
+    inputAmountWithFee: isExactInput ? inputAmount.subtract(feeAmount) : inputAmount.add(feeAmount),
+    inputAmountWithoutFee: inputAmount,
```

Comment:
Is/was the `inputAmount` always equal to `inputAmountWithoutFee` or is this a new behaviour introduced with fee=0?

Reply:
It's not new, actually it's just a copy-paste from `useTradeExactInWithFee` and `useTradeExactOutWithFee`